test/e2e/framework/helper/validation/certificates/certificates.go

```diff
@@ -37,12 +37,20 @@ import (
 // ValidationFunc describes a Certificate validation helper function
 type ValidationFunc func(certificate *cmapi.Certificate, secret *corev1.Secret) error
 
-// Expect2Or3KeysInSecret checks if the secret resource has the correct amount of fields in the secret data
-func Expect2Or3KeysInSecret(_ *cmapi.Certificate, secret *corev1.Secret) error {
-	if !(len(secret.Data) == 2 || len(secret.Data) == 3) {
-		return fmt.Errorf("Expected 2 or 3 keys in certificate secret, but there was %d", len(secret.Data))
+// ExpectValidKeysInSecret checks that the secret contains valid keys
+func ExpectValidKeysInSecret(_ *cmapi.Certificate, secret *corev1.Secret) error {
+	validKeys := [5]string{corev1.TLSPrivateKeyKey, corev1.TLSCertKey, cmmeta.TLSCAKey, "key.der", "tls.pem"}
+	for k := range secret.Data {
+		found := false
+		for _, k2 := range validKeys {
+			if k == k2 {
+				found = true
+			}
+		}
+		if !found {
+			return fmt.Errorf("Expected secret key %s to be a valid key (%v)", k, validKeys)
+		}
 	}
```

Comment:
This looks like it has changed the intention and behavior of this function. This will now return `nil`, for example, if there are _no_ keys in the Secret Data.

We should _at least_ still be catching cases where there are less or more keys than expected, not just that the keys which do exist are a member of the valid set.

Reply:
I agree, The test should fail if there is less thant 2 keys. Because of additional key format feature, it may have a lot of keys in secret. For example if I enable der and pem format I have 5 keys in my secret (tls.key, tls.crt, ca.crt, tls.der, tls.pem)